### PR TITLE
CAT-49 : Remove unapproved GitHub Action use

### PR DESCRIPTION
**Pull request by lainekendall**
Why this PR is needed
----
I didn't realize we were only using approved GitHub actions at EH and I used an unapproved Action. The list of approved actions is [here](https://energyhub.atlassian.net/wiki/spaces/SEC/pages/2996862977/GitHub+Actions#Policy).

Jira ticket reference : [CAT-49](https://energyhub.atlassian.net/browse/CAT-49)

What this PR includes
----
- Remove use of unapproved GitHub Action
- Replace with manual copying of m2 settings file

How to test / verify these changes
----
This was tested in the [jmt](https://github.com/energyhub/java-microservice-template/actions/runs/3624116034/jobs/6110733613).

### CODE REVIEW

File: .github/actions/maven-setup/action.yml [hunk 1]
```diff
@@ -15,54 +15,17 @@ runs:
...
       with:
-        java-version: '17'
-        distribution: 'temurin'
+        java-version: "17"
+        distribution: "temurin"
     - name: Cache local Maven repository
       uses: actions/cache@v3
       with:
-        path: ~/.m2/repository
+        path: $HOME/.m2/repository
```
Comment (lainekendall) [inline .github/actions/maven-setup/action.yml:23]: for consistency

File: .github/actions/maven-setup/action.yml [hunk 2]
```diff
@@ -15,54 +15,17 @@ runs:
...
-            {
-              "id": "snapshots",
-              "username": "${{ inputs.jfrog-user }}",
-              "password": "${{ inputs.jfrog-password }}"
-            }
-          ]
+      shell: bash
+      run: |
+        sed "s/{username}/${{ inputs.jfrog-user }}/;s/{access_token}/${{ inputs.jfrog-password }}/"\
+          ${GITHUB_ACTION_PATH}/home_m2_settings.xml.template > $HOME/.m2/settings.xml
```
Comment (lainekendall) [inline .github/actions/maven-setup/action.yml:31]: copied from @dnrce 's gh action step [here](https://github.com/energyhub/edx/pull/3958/files#diff-b803fcb7f17ed9235f1e5cb1fcd2f5d3b2838429d4368ae4c57ce4436577f03fR51)

File: .github/actions/maven-setup/home_m2_settings.xml.template [hunk 3]
```diff
@@ -0,0 +1,58 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (lainekendall) [inline .github/actions/maven-setup/home_m2_settings.xml.template:1]: copied from https://github.com/energyhub/edx/blob/main/etc/ci/home_m2_settings.xml.template

**Review by eh-yedidovich — APPROVED** [review]
lgtm

**Comment by pcalouche** [comment]
Nice update.  I know I'm late responding to this, but Maven allows environment variables to be used in `settings.xml` as well.  https://medium.com/@clarkbw/environment-variables-for-maven-settings-8280d8a809f9 for reference.  So something like this could work if you setup environments variables for the GH secrets and you would not need to use a `sed` command:

```xml
<server>
      <username>{env.username}</username>
      <password>{env.access_token}</password>
      <id>central</id>
</server>
```

I'm fine either  way.

**Pull request merged**